File: mRemoteV1/Root/Root.Info.vb
Imports System.ComponentModel
Imports mRemote3G.Tools

Namespace Root
    <DefaultProperty("Name")>
    Public Class Info

#Region "Constructors"

        Public Sub New(rootType As RootType)
            Type = rootType
        End Sub

#End Region

#Region "Public Properties"

        Private _name As String = Language.Language.strConnections

        <LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1),
            Browsable(True),
            LocalizedAttributes.LocalizedDefaultValue("strConnections"),
            LocalizedAttributes.LocalizedDisplayName("strPropertyNameName"),
            LocalizedAttributes.LocalizedDescription("strPropertyDescriptionName")>
        Public Overridable Property Name As String
            Get
                Return _name
            End Get
            Set
                If _name = value Then Return
                _name = value
                If TreeNode IsNot Nothing Then
                    TreeNode.Name = value
                    TreeNode.Text = value
                End If
            End Set
        End Property

        <LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1),
            Browsable(True),
            LocalizedAttributes.LocalizedDisplayName("strPasswordProtect"),
            TypeConverter(GetType(Misc.YesNoTypeConverter))>
        Public Property Password As Boolean

        <Browsable(False)>
        Public Property PasswordString As String

        <Browsable(False)>
        Public Property Type As RootType

        <Browsable(False)>
        Public Property TreeNode As TreeNode

#End Region

#Region "Public Enumerations"

        Public Enum RootType
            Connection
            Credential
            PuttySessions
        End Enum

#End Region
    End Class
End Namespace

File: mRemoteV1/Tree/Root/Root.Info.cs
using mRemoteNG.Tools;
using System.ComponentModel;
using System.Windows.Forms;


namespace mRemoteNG.Root
{
	[DefaultProperty("Name")]
    public class Info
    {
        #region Private Properties
        private string _name;
        #endregion

        #region Constructors
        public Info(RootType rootType)
		{
			_name = My.Language.strConnections;
			Type = rootType;
		}
        #endregion
			
        #region Public Properties
        [LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1), 
            Browsable(true),
            LocalizedAttributes.LocalizedDefaultValue("strConnections"),
            LocalizedAttributes.LocalizedDisplayName("strPropertyNameName"),
            LocalizedAttributes.LocalizedDescription("strPropertyDescriptionName")]
        public virtual string Name
		{
			get { return _name; }
			set
			{
				if (_name == value)
				{
					return ;
				}
				_name = value;
				if (TreeNode != null)
				{
					TreeNode.Name = value;
					TreeNode.Text = value;
				}
			}
		}

        [LocalizedAttributes.LocalizedCategory("strCategoryDisplay", 1),
            Browsable(true),
            LocalizedAttributes.LocalizedDisplayName("strPasswordProtect"),
            TypeConverter(typeof(Tools.Misc.YesNoTypeConverter))]
        public bool Password { get; set; }
			
		[Browsable(false)]
        public string PasswordString {get; set;}
			
		[Browsable(false)]
        public RootType Type {get; set;}
			
		[Browsable(false)]
        public TreeNode TreeNode {get; set;}
        #endregion
			
        #region Public Enumerations
		public enum RootType
		{
			Connection,
			Credential,
			PuttySessions
		}
        #endregion
	}
}